src/App.js

```diff
@@ -1,30 +1,71 @@
-import { useEffect, useState } from "react"
-import { Link, Route, Routes } from "react-router-dom"
-import ContactsList from "./components/ContactsList"
-import ContactsAdd from "./components/ContactsAdd"
-import ContactsView from "./components/ContactsView"
-import "./styles/styles.css"
+import { useEffect, useState } from "react";
+import { Link, Route, Routes } from "react-router-dom";
+import ContactsList from "./components/ContactsList";
+import ContactsAdd from "./components/ContactsAdd";
+import ContactsView from "./components/ContactsView";
+import ContactsEdit from "./components/ContactsEdit";
+import LoadingSpinner from "./components/LoadingSpinner";
+import "./styles/styles.css";
 
-export default function App() {
-  const [contacts, setContacts] = useState([])
-  
-  //TODO: Load all contacts on useEffect when component first renders
+const App = () => {
+  const [contacts, setContacts] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetch("http://localhost:4000/contacts")
+      .then((res) => res.json())
+      .then((contactData) => {
+        setContacts(contactData);
+        setIsLoading(false);
+      });
+  }, []);
 
   return (
     <>
       <nav>
         <h2>Menu</h2>
         <ul>
-          {/* TODO: Make these links */}
-          <li>Contacts List</li>
-          <li>Add New Contact</li>
+          <Link to="/">
+            <li>Contacts List</li>
+          </Link>
+          <Link to="/contacts/add">
+            <li>Add New Contact</li>
+          </Link>
         </ul>
       </nav>
       <main>
         <Routes>
-          {/* TODO: Add routes here  */}
+          <Route
+            path="/"
+            element={
+              isLoading ? (
+                <LoadingSpinner />
+              ) : (
+                <ContactsList contacts={contacts} setContacts={setContacts} />
+              )
+            }
```

Comment:
nice implementation of the spinner extension setting isLoading in useEffect an then conditionally rendering component